synapse/config/key.py

```diff
@@ -217,7 +217,18 @@ def read_signing_keys(self, signing_key_path: str, name: str) -> List[SigningKey
 
         signing_keys = self.read_file(signing_key_path, name)
         try:
-            return read_signing_keys(signing_keys.splitlines(True))
+            loaded_signing_keys = read_signing_keys(
+                [
+                    signing_key_line
+                    for signing_key_line in signing_keys.splitlines(keepends=False)
+                    if signing_key_line.strip()
+                ]
+            )
+
+            if not loaded_signing_keys:
+                raise ConfigError(f"No signing keys in file {signing_key_path}")
+
+            return loaded_signing_keys
```

Comment:
I considered a generator expression but I don't necessarily trust the library to not change in such a way that makes it so the generator gets consumed multiple times — this kind of thing has bitten before, so for this tiny file I thought it would be better to just be a bit defensive.

+1 to wondering why keepends=True before. `.split()` knocks them out in any case, but I find it weird